src/components/CartItem.tsx

```diff
@@ -0,0 +1,96 @@
+/* eslint-disable max-len */
+import { getFullPrice } from '../helpers/getFullPrice';
+import { Product } from '../types/Product';
+
+interface Props<T> {
+  product: Product,
+  save: (value: T) => void,
+  products: Product[],
+  setTotalPrice: (value: number) => void,
+  totalPrice: number,
+}
+
+enum Action { add, substract }
+
+export const CartItem: React.FC<Props<Product[]>> = (
+  {
+    product, save, products, setTotalPrice, totalPrice,
+  },
```

Comment:
```suggestion
  {
    product,
    save,
    products,
    setTotalPrice,
    totalPrice,
  },
```

It's better to format it in this way :arrow_up: